raster/r.geomorphon/main.c

```diff
@@ -309,6 +455,23 @@ int main(int argc, char **argv)
         G_verbose_message("Flat threshold distance m: %f, height: %f",
                           flat_distance, flat_threshold_height);
         G_verbose_message("%s version", (extended) ? "Extended" : "Basic");
+        /*
+         * FIXME: It would be nice to have an additional "-s" flag to set the
+         * current computational region just large enough for the one-off
+         * computation (plus some margins and any required outward alignment
+         * to the cell boundary). And also perhaps another "-r" flag to
+         * restore the region afterwards.
+         */
```

Comment:
`calc_pattern()` always looks into the centre of a region cell. As far as I understand it, in this case the best approach is to align the region cells with the raster cells to avoid resampling, but the alignment logic in `g.region` does not seem to be available as a library function. In order to achieve the effect without duplicating the code, the first version of Geomorphon Profiler just calls `g.region` before `r.geomorphon`. This could be a valid improvement for the next version.